src/biguint.rs

```diff
@@ -518,46 +518,50 @@ impl<'a> AddAssign<&'a BigUint> for BigUint {
 
 promote_unsigned_scalars!(impl Add for BigUint, add);
 promote_unsigned_scalars_assign!(impl AddAssign for BigUint, add_assign);
-forward_all_scalar_binop_to_val_val_commutative!(impl Add<BigDigit> for BigUint, add);
-forward_all_scalar_binop_to_val_val_commutative!(impl Add<DoubleBigDigit> for BigUint, add);
+forward_all_scalar_binop_to_val_val_commutative!(impl Add<u32> for BigUint, add);
+forward_all_scalar_binop_to_val_val_commutative!(impl Add<u64> for BigUint, add);
+#[cfg(has_i128)]
+forward_all_scalar_binop_to_val_val_commutative!(impl Add<u128> for BigUint, add);
 
-impl Add<BigDigit> for BigUint {
+impl Add<u32> for BigUint {
     type Output = BigUint;
 
     #[inline]
-    fn add(mut self, other: BigDigit) -> BigUint {
+    fn add(mut self, other: u32) -> BigUint {
         self += other;
         self
     }
 }
-impl AddAssign<BigDigit> for BigUint {
+
+impl AddAssign<u32> for BigUint {
     #[inline]
-    fn add_assign(&mut self, other: BigDigit) {
+    fn add_assign(&mut self, other: u32) {
         if other != 0 {
             if self.data.len() == 0 {
                 self.data.push(0);
             }
 
-            let carry = __add2(&mut self.data, &[other]);
+            let carry = __add2(&mut self.data, &[other as BigDigit]);
```

Comment:
I think there are a bunch of places in here already that used `as` instead of `from/into`. Should I just convert all of them, or is there any potential drawback?

Reply:
Hmm, we can leave that for future cleanup, but let's at least do that before (or during) the `BigDigit` upgrade.

Reply:
created a tracking issue for this: https://github.com/rust-num/num-bigint/issues/66